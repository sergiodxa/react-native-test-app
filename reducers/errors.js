import { Map as map } from 'immutable';

function errors(state = map(), action = {}) {
  switch (action.type) {
    case 'SET_ERROR': {
      return state.set(action.payload.id, action.payload.message);
    }
    case 'REMOVE_ERROR': {
      return state.remove(action.payload.id);
    }
    default: {
      return state;
    }
  }
}

export default errors;
