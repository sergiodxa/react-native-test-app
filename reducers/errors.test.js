import errors from './errors';

const id = 'cj2dpiavu00002mvdtmcx6ucw';

const setAction = {
  type: 'SET_ERROR',
  payload: {
    id,
    message: 'Failed to fetch from API',
  },
};

const removeAction = {
  type: 'REMOVE_ERROR',
  payload: { id },
};


describe('errors reducer', () => {
  let state;

  beforeAll(() => {
    state = errors();
  });

  it('should set a new error message', () => {
    state = errors(state, setAction);
    expect(state).toMatchSnapshot();
  });

  it('should remove an error message', () => {
    state = errors(state, removeAction);
    expect(state).toMatchSnapshot();
  });
});
