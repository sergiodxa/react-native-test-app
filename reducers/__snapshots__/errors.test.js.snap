// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors reducer should remove an error message 1`] = `Object {}`;

exports[`errors reducer should set a new error message 1`] = `
Object {
  "cj2dpiavu00002mvdtmcx6ucw": "Failed to fetch from API",
}
`;
